src/main/java/org/prebid/server/bidder/admixer/AdmixerBidder.java

```diff
@@ -85,23 +85,34 @@ private ExtImpAdmixer parseImpExt(Imp imp) {
         } catch (IllegalArgumentException e) {
             throw new PreBidException(String.format("Wrong Admixer bidder ext in imp with id : %s", imp.getId()));
         }
-        if (StringUtils.length(extImpAdmixer.getZone()) != 36) {
+        if (StringUtils.length(extImpAdmixer.getZone()) < 32 || StringUtils.length(extImpAdmixer.getZone()) > 36) {
             throw new PreBidException("ZoneId must be UUID/GUID");
         }
 
         return extImpAdmixer;
     }
 
     private Imp processImp(Imp imp, ExtImpAdmixer extImpAdmixer) {
-        final Double extImpFloor = extImpAdmixer.getCustomFloor();
-        final BigDecimal customFloor = extImpFloor != null ? BigDecimal.valueOf(extImpFloor) : BigDecimal.ZERO;
+        final BigDecimal bidFloor = calculateBifLoor(extImpAdmixer.getCustomFloor(), imp.getBidfloor());
+
         return imp.toBuilder()
                 .tagid(extImpAdmixer.getZone())
-                .bidfloor(customFloor)
+                .bidfloor(bidFloor)
                 .ext(makeImpExt(extImpAdmixer.getCustomParams()))
                 .build();
     }
 
+    private BigDecimal calculateBifLoor(BigDecimal customFloor, BigDecimal bidFloor) {
+        customFloor = isValidBidFloor(customFloor) ? customFloor : null;
```

Comment:
Create new `final` variable `resolvedCustomBidFloor`. Do not override input args.